meta-refkit/recipes-image/images/refkit-installer-image.bb

```diff
@@ -0,0 +1,188 @@
+                partition=$i
+            else
+                fatal "$output*$gdisk_pnum not found in $output"
+            fi
+        done
+        if [ ! "$partition" ]; then
+            fatal "could not identify partition #$gdisk_pnum in $output"
+        fi
+
+        if [ "$uuid" ]; then
+            # Assume that there's only one ext4 partition and it contains root fs (/)
+            if ! execute mkfs.ext4 -q -v -F -U "$uuid" "$partition"; then
+                fatal "formatting target rootfs partition $gdisk_pnum failed"
+            fi
+            if ! execute mount -t ext4 "$partition" "$output_mountpoint"; then
+                fatal "mounting target rootfs failed"
+            else
+                output_mounted=1
+            fi
+            if ! execute rsync -aAX "$rootfs/" "$output_mountpoint/"; then
+                fatal "copying rootfs failed"
+            fi
+        else
+            if ! execute mkfs.fat "$partition"; then
+                fatal "formating vfat partition $gdisk_pnum failed"
+            fi
+            if ! execute mount -t vfat "$partition" "$output_mountpoint"; then
+                fatal "mounting target vfat partition $gdisk_pnum failed"
+            else
+                output_mounted=1
+            fi
+            if ! execute cp -r "$rootfs/boot/EFI_internal_storage" "$output_mountpoint/EFI"; then
+                fatal "copying EFI files failed"
+            fi
+        fi
+        if ! sync; then
+            fatal "syncing data failed"
+        fi
+        cleanup_populate
+    }
+
+    install_image () {
+        input="${INSTALLER_IMAGE_DATADIR}/$CHOSEN_INPUT"
+        output="/dev/$CHOSEN_OUTPUT"
+        info "Installing from $input to $output."
+
+        confirm_install || return 1
+
+        input_mountpoint=
+        input_mounted=
+
+        cleanup_install_image () {
+            [ "$input_mounted" ] && umount "$input_mountpoint"
+            [ "$input_mountpoint" ] && rmdir "$input_mountpoint"
+            [ "$input" ] && kpartx -d "$input"
+            remove_cleanup cleanup_install_image
+        }
+        add_cleanup cleanup_install_image
+
+        # Assume that there's only one ext4 partition at the end and it
+        # contains the systems' rootfs.
+        loopdev=$(execute kpartx -sav "$input" | tail -1 | sed -e 's/^\(add map \)*\([^ ]*\).*/\2/')
+        if [ ! "$loopdev" ]; then
+            fatal "kpartx failed for $input"
+        fi
+        if ! input_mountpoint=$(mktemp -d); then
+            fatal "could not create mount point"
+        fi
+        if ! execute mount "/dev/mapper/$loopdev" "$input_mountpoint"; then
+            fatal "count not mount rootfs from /dev/mapper/$loopdev"
+        else
+            input_mounted=1
+        fi
+
+        # Clear all partition data on disk
+        if ! execute sgdisk -o "$output"; then
+            fatal "sgdisk $output has failed - damaged disk?"
+        fi
+
+        # Read partition description from rootfs.
```

Comment:
True, it cannot be guaranteed and we should not make that limitation.  I had not checked this carefully enough and still had the mindset (don't know where that came to me) that the installer image had just the rootfs'es of the other images in it (and not the full images).

Anyway, the likely mismatch between ```DSK_IMAGE_FORMAT``` and what's in .wks.in is a problem. How about if do_uefiapp copies the "internal storage" .efi in ```EFI_<PARTUUID>``` and the rootfs PARTUUID is parsed from it.

With, e.g., ```sgdisk -p <wic>``` we can get the same partition data and the partitions carry the filesystem info.